packages/twenty-front/src/modules/ui/field/display/components/LinksDisplay.tsx

```diff
@@ -1,50 +1,43 @@
 import { useMemo } from 'react';
 
+import { getFieldLinkDefinedLinks } from '@/object-record/record-field/meta-types/input/utils/getFieldLinkDefinedLinks';
 import { FieldLinksValue } from '@/object-record/record-field/types/FieldMetadata';
 import { ExpandableList } from '@/ui/layout/expandable-list/components/ExpandableList';
-import { checkUrlType } from '~/utils/checkUrlType';
 import {
   getAbsoluteUrlOrThrow,
   getUrlHostnameOrThrow,
   isDefined,
 } from 'twenty-shared/utils';
 import { LinkType, RoundedLink, SocialLink } from 'twenty-ui/navigation';
+import { checkUrlType } from '~/utils/checkUrlType';
 
 type LinksDisplayProps = {
   value?: FieldLinksValue;
 };
 
 export const LinksDisplay = ({ value }: LinksDisplayProps) => {
-  const links = useMemo(
-    () =>
-      [
-        value?.primaryLinkUrl
-          ? {
-              url: value.primaryLinkUrl,
-              label: value.primaryLinkLabel,
-            }
-          : null,
-        ...(value?.secondaryLinks ?? []),
-      ]
-        .filter(isDefined)
-        .map(({ url, label }) => {
-          let absoluteUrl = '';
-          let hostname = '';
-          try {
-            absoluteUrl = getAbsoluteUrlOrThrow(url);
-            hostname = getUrlHostnameOrThrow(absoluteUrl);
-          } catch {
-            absoluteUrl = '';
-            hostname = '';
-          }
-          return {
-            url: absoluteUrl,
-            label: label || hostname,
-            type: checkUrlType(absoluteUrl),
-          };
-        }),
-    [value?.primaryLinkLabel, value?.primaryLinkUrl, value?.secondaryLinks],
-  );
+  const links = useMemo(() => {
+    if (!isDefined(value)) {
```

Comment:
this would avoid this

Reply:
I can't make the `value` prop of the `LinksDisplay` component required as it's used in the `LinksFieldDisplay` where the `fieldValue` can be `undefined`.

![CleanShot 2025-05-20 at 11 43 32@2x](https://github.com/user-attachments/assets/0d8076ac-69e7-41f3-b023-a6fbe0ae20fe)